### ISSUE TITLE
VsTest task fails fairly frequently with System.NullReferenceException during PublishTestResultPhase

### ISSUE THREAD

**Issue by Adebeer** [microsoft/azure-pipelines-tasks#12701]
Last couple of days we seem to have to re-run our build pipelines quite often due to the VsTest task failing during publish test results step.

## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  Bug

**Enter Task Name**: VsTest

## Environment
- Azure Pipelines  
- alsdevelopmentna.visualstudio.com/AngleNextGen%20Core%20Product%20Pipeline
- Build 20200416.1

- Agent - **Hosted** 
    
```
Pool: Azure Pipelines
Image: windows-2019
Agent: Hosted Agent
```

## Issue Description

Below is error that we've had +3 times this week - have to re-run the pipeline to resolve
Occurs after tests pass and during **MS.VS.TestService.VstestConsoleAdapter.PublishTestResultPhase** step with  **System.NullReferenceException**

### Task logs
[vsTestFail.txt](https://github.com/microsoft/azure-pipelines-tasks/files/4484856/vsTestFail.txt)

### Error logs

```
##[debug]Current Phase: MS.VS.TestService.VstestConsoleAdapter.PublishTestResultPhase Phase Result: MS.VS.TestService.VstestConsoleAdapter.PublishTestResultPhaseResults
2020-04-16T03:32:57.0902652Z ##[error]Error occurred while publishing test results : System.AggregateException: One or more errors occurred. ---> System.NullReferenceException: Object reference not set to an instance of an object.
2020-04-16T03:32:57.0905340Z ##[debug]Processed: ##vso[task.logissue type=error;]Error occurred while publishing test results : System.AggregateException: One or more errors occurred. ---> System.NullReferenceException: Object reference not set to an instance of an object.
```

**Comment by ShreyasRmsft** [comment]
We've fixed this and the fix should be available in task version 2.168.0 or higher, the deployment should reach all accounts in a few working days.

**Comment by Adebeer** [comment]
Thanks - haven't noticed this error anymore.
You can probably close this issue - although bit disappointed as, since yesterday, getting new error popping up (#12808)
